src/kv/memstore.rs

```diff
@@ -128,137 +128,24 @@ impl Write for WriteTransaction<'_> {
-        assert!(!ms.has("foo").await?);
-        assert_eq!(None, ms.get("foo").await?);
-
-        ms.put("foo", b"bar").await?;
-        assert!(ms.has("foo").await?);
-        assert_eq!(Some(b"bar".to_vec()), ms.get("foo").await?);
-
-        ms.put("foo", b"baz").await?;
-        assert!(ms.has("foo").await?);
-        assert_eq!(Some(b"baz".to_vec()), ms.get("foo").await?);
-
-        assert!(!ms.has("baz").await?);
-        assert_eq!(None, ms.get("baz").await?);
-        ms.put("baz", b"bat").await?;
-        assert!(ms.has("baz").await?);
-        assert_eq!(Some(b"bat".to_vec()), ms.get("baz").await?);
-
-        Ok(())
-    }
-
-    #[async_std::test]
-    async fn test_read_transaction() -> std::result::Result<(), StoreError> {
-        let mut ms = MemStore::new();
-        ms.put("k1", b"v1").await?;
-
-        let rt = ms.read().await?;
-        assert!(rt.has("k1").await?);
-        assert_eq!(Some(b"v1".to_vec()), rt.get("k1").await?);
-
-        Ok(())
-    }
-
-    #[async_std::test]
-    async fn test_write_transaction() -> std::result::Result<(), StoreError> {
-        let mut ms = MemStore::new();
-        ms.put("k1", b"v1").await?;
-        ms.put("k2", b"v2").await?;
-
-        // Test put then commit.
-        let wt = ms.write().await?;
-        assert!(wt.has("k1").await?);
-        assert!(wt.has("k2").await?);
-        wt.put("k1", b"overwrite").await?;
-        wt.commit().await?;
-        assert_eq!(Some(b"overwrite".to_vec()), ms.get("k1").await?);
-        assert_eq!(Some(b"v2".to_vec()), ms.get("k2").await?);
-
-        // Test put then rollback.
-        let wt = ms.write().await?;
-        wt.put("k1", b"should be rolled back").await?;
-        wt.rollback().await?;
-        assert_eq!(Some(b"overwrite".to_vec()), ms.get("k1").await?);
-
-        // Test del then commit.
-        let wt = ms.write().await?;
-        wt.del("k1").await?;
-        assert!(!wt.has("k1").await?);
-        wt.commit().await?;
-        assert!(!ms.has("k1").await?);
-
-        // Test del then rollback.
-        assert_eq!(true, ms.has("k2").await?);
-        let wt = ms.write().await?;
-        wt.del("k2").await?;
-        assert!(!wt.has("k2").await?);
-        wt.rollback().await?;
-        assert!(ms.has("k2").await?);
-
-        // Test overwrite multiple times then commit.
-        let wt = ms.write().await?;
-        wt.put("k2", b"overwrite").await?;
-        wt.del("k2").await?;
-        wt.put("k2", b"final").await?;
-        wt.commit().await?;
-        assert_eq!(Some(b"final".to_vec()), ms.get("k2").await?);
-
-        // Test as_read.
-        let wt = ms.write().await?;
-        wt.put("k2", b"new value").await?;
-        let rt = wt.as_read();
-        assert!(rt.has("k2").await?);
-        assert_eq!(Some(b"new value".to_vec()), rt.get("k2").await?);
+    async fn test_memstore() -> std::result::Result<(), StoreError> {
+        let ms = &mut MemStore::new();
```

Comment:
Feels like it will be easy to accidentally forget to add to this list. Can it be mechanised more?

Reply:
yes indeed. lets sort out what we're doing with this PR then i will address.